packages/framework/ai-collab/README.md

```diff
@@ -238,10 +239,166 @@ To see detailed information about each event, please read their cooresponding [t
 
 
 ### Using Trace Id's
+
 Debug Events in ai-collab have two different types of trace id's:
 - `traceId`: This field exists on all debug events and can be used to correlate all debug events that happened in a single execution of `aiCollab()`. Sorting the events by timestamp will show the proper chronological order of the events. Note that the events should already be emitted in chronological order.
 - `eventFlowTraceId`: this field exists on all `EventFlowDebugEvents` and can be used to correlate all events from a particular event flow. Additionally all LLM api call events will contain the `eventFlowTraceId` field as well as a `triggeringEventFlowName` so you can link LLM API calls to a particular event flow.
 
+## Edit Difference Visualizations
+
+ai-collab provides an array of `Diff` objects with its response. Each of these objects allows developers to identify tree nodes that have been modified as a result of ai collaboration and visualize them according to their needs.
+
+Every `Diff` will include one or more `NodePaths` representing the nodes affected by a single edit created by the ai agent. A `NodePath` is an array whose items represent segment paths, beginning from the node targeted for modification (at the start of the array) all the way back to the root node passed to the ai-collab function call (at the end of the array), along with an explanation directly from the ai agent as to why it performed an edit.
+
+Let's take a look at some examples for the following SharedTree application schema
+
+```ts
+import { aiCollab, DebugEvent, Diff } from "@fluidframework/ai-collab/alpha";
+
+const sf = new SchemaFactory("testApp");
+
+class Todo extends sf.object("Todo", {
+	id: sf.identifier,
+	title: string
+	description: string
+}) {}
+
+class TestAppRootObject extends sf.object("TestAppRootObject", {
+	id: sf.identifier,
+	todos: sf.array([Todo]),
+	innerObject: sf.object("InnerObject", {
+		nestedTodos: sf.array([Todo]),
+	}),
+}) {}
+
+const response = aiCollab({
+	openAI: {
+		client: new OpenAI({
+			apiKey: OPENAI_API_KEY,
+		}),
+		modelName: "gpt-4o",
+	},
+	treeNode: view.root,
+	prompt: {
+		systemRoleContext:
+			"You are a manager that is helping out with a project management tool. You have been asked to edit a group of tasks.",
+		userAsk: userAsk,
+	},
+	limiters: {
+		maxModelCalls: 25
+	}
+	planningStep: true,
+	finalReviewStep: true,
+	debugEventLogHandler: (event: DebugEvent) => {console.log(event);}
+});
+
+const Diffs: diff[] = response.Diffs
+```
+
+Each Diff will contain one or more `NodePath`'s. Each `NodePath` provides an array of objects that detail the path from the root node passed to ai-collab, down to the node targeted for editing. The first index in the `NodePath` is an object pointing to the target node and the last index is always the root node.
+
+Lets look at an example of the Insert Diff
```

Comment:
Nit: This and the following paragraph should probably go inside the `### Example Insert Ui Diff` subheading.